13-offer-encoding.md

```diff
@@ -0,0 +1,589 @@
+        * [`sha256`:`offer_id`]
+    1. type: 8 (`amount`)
+    2. data:
+        * [`tu64`:`amount`]
+	1. type: 10 (`description`)
+    2. data:
+		* [`...*byte`:`description`]
+    1. type: 12 (`features`)
+    2. data:
+        * [`...*byte`:`features`]
+    1. type: 16 (`blindedpath`)
+    2. data:
+        * [`point`:`blinding`]
+        * [`...*onionmsg_path`:`path`]
+    1. type: 18 (`blindedpay`)
+    2. data:
+        * [`...*blinded_payinfo`:`payinfo`]
+	1. type: 20 (`vendor`)
+    2. data:
+		* [`...*byte`:`vendor`]
+    1. type: 30 (`node_id`)
+    2. data:
+        * [`pubkey32`:`node_id`]
+	1. type: 32 (`quantity`)
+    2. data:
+		* [`tu64`:`quantity`]
+	1. type: 34 (`refund_for`)
+    2. data:
+        * [`sha256`:`refunded_payment_hash`]
+    1. type: 38 (`payer_key`)
+    2. data:
+        * [`pubkey32`:`key`]
+    1. type: 40 (`timestamp`)
+    2. data:
+        * [`tu32`:`timestamp`]
+    1. type: 42 (`payment_hash`)
+    2. data:
+        * [`sha256`:`payment_hash`]
+    1. type: 44 (`expiry`)
+    2. data:
+        * [`tu32`:`expiry_seconds`]
+    1. type: 46 (`cltv`)
+    2. data:
+        * [`tu32`:`min_final_cltv_expiry`]
+    1. type: 48 (`fallbacks`)
+    2. data:
+	    * [`u8`:`num`]
+		* [`num*fallback_address`:`fallbacks`]
+    1. type: 50 (`routehelp`)
+    2. data:
+		* [`...*payroute`:`routes`]
+	1. type: 52 (`refund_signature`)
+    2. data:
+        * [`signature`:`payer_signature`]
+    1. type: 240 (`signature`)
+    2. data:
+        * [`signature`:`sig`]
+
+1. subtype: `payroute`
+2. data:
+   * [`point`:`blinding`]
+   * [`u16`:`num_hops`
+   * [`onionmsg_path`:`num_hops*path`]
+   * [`blinded_payinfo`:`num_hops*payinfo`]
+
+1. subtype: `blinded_payinfo`
+2. data:
+   * [`u32`:`fee_base_msat`]
+   * [`u32`:`fee_proportional_millionths`]
+   * [`u16`:`cltv_expiry_delta`]
+   * [`u16`:`flen`]
+   * [`flen*byte`:`features`]
+
+1. subtype: `fallback_address`
+2. data:
+   * [`byte`:`type`]
+   * [`u16`:`len`]
+   * [`len*byte`:`address`]
+
+## Requirements
+
+A writer of an invoice:
+  - if the `invrequest` contained the same `offer_id`, `payer_key` and `invrequest_recurrence` (if any) as a previous `invrequest`:
+    - MAY simply reuse the previous invoice.
+  - otherwise:
+    - MUST NOT reuse a previous invoice.
+  - MUST set `timestamp` to the number of seconds since Midnight 1
+    January 1970, UTC.
+  - MUST set `payment_hash` to the SHA2 256-bit hash of the
+    `payment_preimage` that will be given in return for payment.
+  - MUST set `node_id` to the public key of the node to pay the invoice to.
+  - MUST specify exactly one signature TLV: `signature`.
+    - MUST set `sig` to the signature using `node_id` as described in [Signature Calculation](#signature-calculation).
+  - if the chain for the invoice is not solely bitcoin:
+    - MUST specify `chains` the offer is valid for.
+  - otherwise:
+    - the bitcoin chain is implied as the first and only entry.
+  - if it supports bolt11 features:
```

Comment:
nit:
```suggestion
  - if it supports bolt9 features:
```

Reply:
I changed this to "has bolt9 features" since I figure we should make support compulsory.